<!DOCTYPE html>
<html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    
    <title>Dr. Beibei Ru</title>
    <link href="./css" rel="stylesheet" type="text/css">
    <link type="text/css" rel="stylesheet" href="./bootstrap.min.css">
    <link type="text/css" rel="stylesheet" href="./style.css">
</head>
<body data-spy="scroll" data-target="#header" style="">
<div id="header" class="navbar-fixed-top navbar navbar-default navbar-inverse" role="navigation">
    <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        </button>
        <a href="https://beibeiru.github.io/" class="navbar-brand">Dr. Beibei Ru</a>
    </div>
    <div class="collapse navbar-collapse">
    <ul class="nav navbar-nav navbar-right">
        <li class="active"><a href="https://beibeiru.github.io/#about" style="color: #1da1f2; font-weight: bold">about me</a></li>
        <li class=""><a href="https://beibeiru.github.io/#research">research</a></li>
        <li class=""><a href="https://beibeiru.github.io/#contact">contact</a></li>
    </ul>
    </div>
</div>

<div class="container">
<section id="about">
<img src="./xx.jpg" id="imgme" class="img-rounded">
<div>
    <p>
      beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru   
    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru   
    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru   
    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru   
    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru beibei ru 
	    beibei ru beibei ru beibei ru beibei ru 
    </p>
</div>
</section>

<section id="education">
<h3>Education</h3>
<div>
    
    <table class="table">
      <thead>
        <tr>
          <th>Year</th>
          <th>Degree</th>
          <th>Place</th>
        </tr>
      </thead>
      <tbody>

        <tr>
          <th>2018</th>
          <td>Ph.D.</td>
          <td>The University of Hong Kong</td>
        </tr>
        <tr>
          <th>2013</th>
          <td>M.Sc.</td>
          <td>University of Electronic Science and Technology of China</td>
        </tr>
        <tr>
          <th>2010</th>
          <td>B.Sc.</td>
          <td>University of Electronic Science and Technology of China</td>
        </tr>
      </tbody>
    </table>

</div>
</section>

<section id="research"> 
<h3>Research</h3>

<section>
<h4>Projects</h4>
<table class="table">
<tbody>
<tr><th>Snakemake</th><td>
<p>Snakemake is a workflow engine and language. It aims to reduce the complexity of creating workflows by providing a fast and comfortable execution environment, together with a clean and modern domain specific specification language (DSL) in python style.</p>
<p class="text-right"><a href="https://snakemake.readthedocs.io/">visit homepage</a></p>
</td></tr>
<tr><th>MERFISHtools</th><td>
<p>A Bayesian model for single-cell transcript (differential) expression analysis on <a href="https://science.sciencemag.org/content/early/2015/04/08/science.aaa6090.abstract">MERFISH</a> data.
The model allows to overcome systematic biases occurring with MERFISH and provides measures of uncertainty
and control of the false discovery rate in a strictly Bayesian way. MERFISHtools is a corresponding command
line client and analysis library written in <a href="https://rust-lang.org/">Rust</a> and <a href="https://python.org/">Python</a>.
MERFISHtools is also available via <a href="https://bioconda.github.io/">Bioconda</a>.</p>
<p class="text-right"><a href="https://merfishtools.github.io/">visit homepage</a></p>
</td></tr>
<tr><th>Bioconda</th><td>
<p>A distribution of bioinformatics software realized as a channel for the versatile package manager <a href="https://conda.pydata.org/docs">Conda</a>.</p>
<p class="text-right"><a href="https://bioconda.github.io/">visit homepage</a></p>
</td></tr>
<tr><th>Rust-Bio</th><td>
<p>A bioinformatics library written in the <a href="https://rust-lang.org/">Rust language</a>.
The implementation provides state of the art solutions for common tasks in bioinformatics,
focusing on stability by using comprehensive unit tests and continuous integration.</p>
<p class="text-right"><a href="https://rust-bio.github.io/">visit homepage</a></p>
</td></tr>
<tr><th>ALPACA</th><td>
<p>ALPACA is a variant caller for next-generation sequencing data that incorporates sample based filtering into the calling. This allows intuitive control of the false discovery rate with generic
sample filtering scenarios. Further, it uses preprocessing and merging of BCF files to solve the N+1 problem: an existing study can be extended with new samples without redundant computations. After the preprocessing, the actual calling is a matter of seconds.</p>
<p class="text-right"><a href="https://github.com/johanneskoester/alpaca">visit homepage</a></p>
</td></tr>
<tr><th>PEANUT</th><td>
<p>PEANUT is a read mapper for DNA or RNA sequence reads.
By exploiting the massive parallelism of modern graphics processors and a novel index datastructure, PEANUT achieves superior speed compared to current state of the art read mappers like BWA MEM, Bowtie2 and RazerS3, while maintaining their accuracy. It thereby allows to report both only the best hits or all hits of a read. In case of reporting all hits, PEANUT is four to ten times faster than competitors.</p>
<p class="text-right"><a href="https://peanut.readthedocs.org/">visit homepage</a></p>
</td></tr>
<tr><th>LibModalLogic</th><td>
<p>LibModalLogic is a JAVA implementation of Modal Logic K and Propositional Logic. Logic formulas can be build in memory, saved to and read from MathML and formatted human readable. Reasoning is implemented by the (modal) logic tableau algorithm, including dynamic backtracking for maximum performance.</p>
<p class="text-right"><a href="https://libmodallogic.googlecode.com/">visit at google code</a></p>
</td></tr>
<tr><th>TRMiner</th><td>
<p>TRMiner is a python tool that aims at scientific data curators. It allows to rapidly prune large collections of scientific publications to sentences relevant for a given mining goal, using a linear time matching algorithm.</p>
<p class="text-right"><a href="https://trminer.googlecode.com/">visit at google code</a></p>
</td></tr>
<tr><th>Protein Hypernetworks</th><td>
<p>Protein Hypernetworks are an approach for endowing protein networks with interaction dependencies using propositional logic. This allows refined network based predictions of protein complexes, functional importance and functional similarity.</p>
<p class="text-right"><a href="https://proteinhypernetworks.googlecode.com/"> visit at google code</a></p>
</td></tr>
</tbody>
</table>
</section>



<section>
<h4>Talks</h4>
<ul>
  <li><a class="hiddenlink" href="http://slides.com/johanneskoester/aebc-2017">Keynote about workflow and tool management at AEBC 2017, Tübingen.</a></li>
  <li><a class="hiddenlink" href="http://slides.com/johanneskoester/keynote-jk-dresden-2016">Keynote (german) about the in-silico reproducibility crisis at Helmholtz Workshop on sustainable software development, Dresden 2016.</a></li>
  <li><a class="hiddenlink" href="https://slides.com/johanneskoester/rust-bio-mozilla">Invited talk about Rust-Bio at Rust Bay Area Meeting hosted by Mozilla, Los Angeles 2016.</a></li>
  <li><a class="hiddenlink" href="https://slides.com/johanneskoester/prosic-hitseq-2016">Talk about genotyping somatic insertions and deletions at HiTSeq, Orlando 2016.</a></li>
  <li><a class="hiddenlink" href="https://slides.com/johanneskoester/snakemake-bioconda-leeselab">Invited talk about Snakemake and Bioconda at Aquatic Ecosystem Research, University of Duisburg-Essen, Essen 2016.</a></li>
  <li><a class="hiddenlink" href="https://slides.com/johanneskoester/merfishtools">Talk about MERFISHtools at the Life Science seminar, CWI Amsterdam 2016</a></li>
	<li><a class="hiddenlink" href="https://slides.com/johanneskoester/snakemake-broad-2015">Invited talk about Snakemake at the Broad Institute, Boston 2015.</a></li>
	<li><a class="hiddenlink" href="https://slides.com/johanneskoester/deck-1">Snakemake workshop at the German Conference on Bioinformatics, Dortmund 2015.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-dresden-peanut-2014.pdf">Invited talk about PEANUT at the Workshop on Fast Data Processing on GPUs, Dresden CUDA Center of Excellence, Dresden 2014.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-dtl-snakemake-2014.pdf">Invited talk about Snakemake at the DTL Focus Meeting: "NGS Production Pipelines", Dutch Techcentre for Life Sciences, Utrecht 2014.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-ngs-snp-calling-2013.pdf">Guest lecture at the Faculty of Statistics, TU Dortmund, 2013 on "Detecting SNVs with next generation sequencing".</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-retreat-2013.pdf">Northwestern Germany Bioinformatics Retreat 2013, on novel features and algorithmic details of Snakemake.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-gcb-2012.pdf">German Conference on Bioinformatics, September 2012, on "Building and Documenting Bioinformatics Workflows with Python-based Snakemake".</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-ismp-2012.pdf">International Symposium on Mathematical Programming 2012, on protein hypernetworks.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-9th-pathways-2011.pdf">9th International Conference on Pathways, Networks and Systems Medicine on Crete 2011, on protein hypernetworks. It was honored with a Travel Award.</a></li>
  <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/slides/jk-tcs-workshop-2011.pdf">Mini-Workshop Theoretical Computer Science in Dortmund 2011, on theoretical aspects of protein hypernetworks.</a></li>
</ul>
</section>
<section>
<h4>Poster</h4>
<ul>
	<li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/snakemake+bioconda-2016.pdf">ECCB 2016, The Hague, Netherlands. <i>Fully reproducible data analysis with Snakemake and Bioconda</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/pqg-2015.pdf">PQG Conference on Single Cell Genomics, Harvard T.H. Chan School of Public Health 2015. <i>A Bayesian model for gene expression on MERFISH data</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/mco-retreat-2015.pdf">Medical Oncology Retreat, Dana-Farber Cancer Institute, Harvard Medical School 2015. <i>An algebra of variant loci</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/eccb-2014.pdf">European Conference on Computational Biology 2014. <i>Massively parallel read mapping on GPUs with the q-group index and PEANUT.</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/gcb-2014.pdf">German Conference on Bioinformatics 2014. <i>An algebra of single nucleotide variants.</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/forschungstag-2014.pdf">University Hospital Essen, Science Day 2014. <i>An algebra of single nucleotide variants.</i></a>
	</li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/ccss-ude-2013.pptx">Center for Computational Sciences and Simulation, University Duisburg-Essen 2013. <i>DNA-Sequenzanalysen mit GPUs.</i></a>
    </li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/forschungstag-2013.pdf">University Hospital Essen, Science Day 2013. <i>Massively parallel read mapping with GPUs.</i></a>
    </li><li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/jk-qpcr-ngs-freising-2013.pdf">qPCR-NGS Meeting in Freising 2013. <i>Bioinformatics Workflows with Snakemake.</i> </a></li>
    <li><a class="hiddenlink" href="https://johanneskoester.bitbucket.io/posters/jk-hypernetworks-2011.pdf">University Hospital Essen, Science Day 2011. <i>Protein hypernetworks.</i></a></li>
</ul>
</section>
<section>
<h4>Awards and Grants</h4>
<ul>
<li><a href="https://www.cwi.nl/news/2016/veni-grants-l%C3%A9o-ducas-and-johannes-k%C3%B6ster">NWO Veni grant (€ 250,000)</a> for the project "Fully reproducible workflows scaling from workstations to the cloud"</li>
<li><a href="http://www.cs.tu-dortmund.de/nps/de/Aktuelles/Newsarchiv/2011/2011_04_11_Uhde_Preis_2011/index.html">Uhde-Award 2011</a> for my diploma thesis "Propagating Interaction Logic towards Predictive Protein Hypernetworks".</li>
<li>Honorable Mention at the Doktorandenkolleg Ruhr 2011 for my poster "Protein Hypernetworks".</li>
<li>Poster Award of the University Hospital Essen at the Forschunstag 2011 for my poster "Protein Hypernetworks".</li>
<li>Travel Award of the 9th International Conference on Pathways, Networks and Systems Medicine 2011 for my talk on protein hypernetworks.</li>
</ul>
</section>

</section>
	
	
<section>
<h3>Publications</h3>
<ul>
<li>Ru B, Wong CN, Tong Y, Zhong JY, Zhong SSW, Wu WC, Chu KC, Wong CY, Lau CY, Chen I, Chan NW, Zhang J. TISIDB: an integrated repository portal for tumor-immune system interactions. Bioinformatics. 2019, btz210.</li>
</ul>
</section>	
	
	
<section id="contact">
<h3>Contact</h3>
<table class="table">
<tbody>
<tr><th>email</th><td><a href="mailto:johannes.koester@uni-due.de">johannes.koester@uni-due.de</a></td></tr>
<tr><th>phone</th><td>+49 (0)201 723 1908</td></tr>
<tr><th>www</th><td><a href="https://koesterlab.github.io/">https://koesterlab.github.io</a></td></tr>
<tr><th>keybase</th><td><a href="https://keybase.io/johanneskoester">johanneskoester</a></td></tr>
<tr>
    <th>office</th>
    <td>
        Room 1.13
        University Hospital Essen
        Virchowstr. 183
        45147 Essen
    </td>
</tr>
<tr>
    <th>postal</th>
    <td>
        Dr. rer. nat. Johannes Köster<br>
        Institute of Human Genetics<br>
        University of Duisburg-Essen<br>
        Hufelandstr. 55<br>
        45147 Essen<br>
        Germany
    </td>
</tr>
</tbody>
</table>
</section>
</div>
<script src="./jquery.min.js"></script>
<script src="./bootstrap.min.js"></script>


</body></html>
